osu.Game/Tests/Visual/Multiplayer/TestMultiplayerClient.cs

```diff
@@ -282,13 +296,78 @@ public override Task ChangeUserMods(IEnumerable<APIMod> newMods)
             return Task.CompletedTask;
         }
 
+        private CancellationTokenSource? countdownFinishSource;
+        private CancellationTokenSource? countdownStopSource;
+        private Task countdownTask = Task.CompletedTask;
+
+        public void FinishCountdown() => countdownFinishSource?.Cancel();
```

Comment:
I think from what you explained, this causes the countdown to immediately finish *unless* cancelled? Would appreciate xmldoc and potentially renaming to `SkipToEndOfCountdown` or similar.

Reply:
Depends on what you define as finishing a countdown. If your definition of finishing a countdown is running the start match callback, then no. My definition of finishing the countdown is finishing the wait of however many seconds, it doesn't relate to the callback at all.

Before I make any changes, please confirm that you are okay with the following changes:
```diff
diff --git a/osu.Game/Tests/Visual/Multiplayer/TestMultiplayerClient.cs b/osu.Game/Tests/Visual/Multiplayer/TestMultiplayerClient.cs
index 7562b1d0e2..ea991af914 100644
--- a/osu.Game/Tests/Visual/Multiplayer/TestMultiplayerClient.cs
+++ b/osu.Game/Tests/Visual/Multiplayer/TestMultiplayerClient.cs
@@ -296,11 +296,15 @@ public override Task ChangeUserMods(IEnumerable<APIMod> newMods)
             return Task.CompletedTask;
         }

-        private CancellationTokenSource? countdownFinishSource;
+        private CancellationTokenSource? countdownSkipSource;
         private CancellationTokenSource? countdownStopSource;
         private Task countdownTask = Task.CompletedTask;

-        public void FinishCountdown() => countdownFinishSource?.Cancel();
+        /// <summary>
+        /// Skips to the end of the currently-running countdown, if one is running,
+        /// and runs the callback (e.g. to start the match) as soon as possible unless the countdown has been cancelled.
+        /// </summary>
+        public void SkipToEndOfCountdown() => countdownSkipSource?.Cancel();

         public override async Task SendMatchRequest(MatchUserRequest request)
         {
@@ -313,8 +317,8 @@ public override async Task SendMatchRequest(MatchUserRequest request)
                     countdownStopSource?.Cancel();

                     var stopSource = countdownStopSource = new CancellationTokenSource();
-                    var finishSource = countdownFinishSource = new CancellationTokenSource();
-                    var cancellationSource = CancellationTokenSource.CreateLinkedTokenSource(stopSource.Token, finishSource.Token);
+                    var skipSource = countdownSkipSource = new CancellationTokenSource();
+                    var cancellationSource = CancellationTokenSource.CreateLinkedTokenSource(stopSource.Token, skipSource.Token);
                     var countdown = new MatchStartCountdown { TimeRemaining = matchCountdownRequest.Duration };

                     Task lastCountdownTask = countdownTask;
```

Because I intend to verbatim copy this into https://github.com/ppy/osu-server-spectator/pull/109#discussion_r833931538.

Reply:
This part seems fine, yep.